File: server/src/main/java/org/hkurh/doky/users/UserEntityRepository.java
package org.hkurh.doky.users;

import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.CrudRepository;
import org.springframework.lang.NonNull;

public interface UserEntityRepository extends CrudRepository<UserEntity, Long>, JpaSpecificationExecutor<UserEntity> {

    @Query("select u from UserEntity u where u.uid = ?1")
    UserEntity findByUid(@NonNull String uid);
}

File: server/src/main/java/org/hkurh/doky/users/UserEntityRepository.kt
package org.hkurh.doky.users

import org.springframework.data.jpa.repository.JpaSpecificationExecutor
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.CrudRepository

interface UserEntityRepository : CrudRepository<UserEntity?, Long?>, JpaSpecificationExecutor<UserEntity?> {
    @Query("select u from UserEntity u where u.uid = ?1")
    fun findByUid(uid: String): UserEntity?
}
